File: packages/shared/src/types/payment.ts
/**
 * 支付相关类型 — 镜像后端 PaymentInfo / PaymentStatusResult
 */

export interface PaymentInfo {
  paymentId: string;
  method: string;
  amount: string;
  payUrl: string;
}

export interface PaymentRecord {
  id: string;
  method: string;
  amount: string;
  status: string;
  transactionId: string | null;
  createdAt: string;
}

export interface PaymentStatusResult {
  orderId: string;
  orderStatus: string;
  payments: PaymentRecord[];
}

File: apps/android/app/src/main/java/com/example/shop/feature/order/data/model/Payment.kt
package com.example.shop.feature.order.data.model

import kotlinx.serialization.Serializable

// ── Requests ──

@Serializable
data class PaymentCreateRequest(
    val orderId: String,
    val method: String = "mock",
)

@Serializable
data class PaymentQueryRequest(val orderId: String)

@Serializable
data class PaymentNotifyRequest(
    val orderId: String,
    val transactionId: String,
    val status: String,
    val amount: Double,
    val method: String,
)

// ── Responses ──

@Serializable
data class PaymentInfo(
    val paymentId: String,
    val method: String,
    val amount: String,
    val payUrl: String = "",
)

@Serializable
data class PaymentStatusResult(
    val orderId: String,
    val orderStatus: String,
    val payments: List<PaymentRecord> = emptyList(),
)

@Serializable
data class PaymentRecord(
    val id: String,
    val method: String,
    val amount: String,
    val status: String,
    val transactionId: String? = null,
    val createdAt: String = "",
)
